platform-sdk/swirlds-jasperdb/src/test/java/com/swirlds/merkledb/files/DataFileReaderCloseTest.java

```diff
@@ -0,0 +1,139 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.swirlds.merkledb.files;
+
+import com.swirlds.common.io.utility.TemporaryFileBuilder;
+import com.swirlds.merkledb.collections.LongList;
+import com.swirlds.merkledb.collections.LongListOffHeap;
+import com.swirlds.merkledb.serialize.DataItemHeader;
+import com.swirlds.merkledb.serialize.DataItemSerializer;
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.nio.channels.ClosedByInterruptException;
+import java.nio.file.Path;
+import java.util.Random;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicReference;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+class DataFileReaderCloseTest {
+
+    private static DataFileCollection<long[]> collection;
+
+    private static final DataItemSerializer<long[]> serializer = new TwoLongSerializer();
+
+    @BeforeAll
+    static void setup() throws IOException {
+        final Path dir = TemporaryFileBuilder.buildTemporaryFile("readerIsOpenTest");
+        collection = new DataFileCollection<>(dir, "store", serializer, null);
+    }
+
+    @AfterAll
+    static void teardown() throws IOException {
+        collection.close();
+    }
+
+    @Test
+    void readerIsOpenTest() throws Exception {
+        final int COUNT = 100;
+        collection.startWriting();
+        final LongList index = new LongListOffHeap();
+        for (int i = 0; i < COUNT; i++) {
+            index.put(i, collection.storeDataItem(new long[] {i, i + 1}));
+        }
+        //noinspection resource
+        collection.endWriting(0, COUNT - 1);
+        final AtomicBoolean readingThreadStarted = new AtomicBoolean(false);
+        final AtomicBoolean readingThreadInterrupted = new AtomicBoolean(false);
+        final AtomicReference<IOException> exceptionOccurred = new AtomicReference<>();
+        final Thread readingThread = new Thread(() -> {
+            final Random rand = new Random();
+            try {
+                while (true) {
+                    final int i = rand.nextInt(COUNT);
+                    final long dataLocation = index.get(i);
+                    final long[] item = collection.readDataItem(dataLocation);
+                    Assertions.assertEquals(i, item[0]);
+                    Assertions.assertEquals(i + 1, item[1]);
+                    readingThreadStarted.set(true);
+                }
+            } catch (final ClosedByInterruptException e) {
+                // This is expected
+            } catch (final IOException e) {
+                exceptionOccurred.set(e);
+            }
+            readingThreadInterrupted.set(true);
```

Comment:
should this line be in the catch for closed by interrupted exception?

Reply:
I think the var name is confusing, and the var seems redundant at all. Let me re-implement it using `Thread.join()` instead.